src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        return new Ball(colorSupplier.getRandomColor(),random.nextInt(100));
```

Comment:
According to the checklist, we should avoid creating instances of the same class multiple times if one instance can be reused. In this case, the `ColorSupplier` and `Random` instances can be created once and reused for generating multiple balls. Consider moving the instantiation of `ColorSupplier` and `Random` to class-level fields.